File: CPP/Program/Device.cpp
#include "Electronic.cpp"

class Device : public Electronic {
protected:
    string brand;
    int warranty;  // in months
    int power;     // watt

public:
    Device() : Electronic(), brand(""), warranty(0), power(0) {}
    Device(int id, string name, string category, int price, string brand, int warranty, int power)
        : Electronic(id, name, category, price), brand(brand), warranty(warranty), power(power) {}

    void setBrand(string brand) { this->brand = brand; }
    string getBrand() const { return brand; }

    void setWarranty(int warranty) { this->warranty = warranty; }
    int getWarranty() const { return warranty; }

    void setPower(int power) { this->power = power; }
    int getPower() const { return power; }

    void showInfo() override {
        Electronic::showInfo();
        cout << "Brand: " << brand << ", Warranty: " << warranty << " months, Power: " << power << "W";
    }

    ~Device() {}
};

File: Java/Program/Device.java
public class Device extends Electronic {
    private String brand;
    private int warranty;
    private int power;

    public Device() {
        this.brand = "";
        this.warranty = 0;
        this.power = 0;
    }

    public Device(String brand, int warranty, int power) {
        this.brand = brand;
        this.warranty = warranty;
        this.power = power;
    }

    public void setBrand(String brand) { this.brand = brand; }
    public String getBrand() { return this.brand; }

    public void setWarranty(int warranty) { this.warranty = warranty; }
    public int getWarranty() { return this.warranty; }

    public void setPower(int power) { this.power = power; }
    public int getPower() { return this.power; }
}
